crates/wysiwyg/src/composer_model/format.rs

```diff
@@ -51,7 +53,189 @@ where
+                                );
+                            }
+                        }
+                        DomNode::LineBreak(_) => {
+                            // We should split inline code nodes at line breaks
+                            if !cur_text.is_empty() {
+                                nodes_to_add
+                                    .insert(0, DomNode::new_text(cur_text));
+                            }
+                            nodes_to_add.insert(0, DomNode::new_line_break());
+                            self.state.dom.remove(&leaf.node_handle);
+                            // Update insertion point and reset text
+                            insert_text_at = Some(ancestor_child.clone());
+                            cur_text = S::default();
+                        }
+                        _ => panic!(
+                            "Leaf should be either a line break or a text node"
+                        ),
+                    }
+                }
+
+                // Insert the nodes into the Dom inside an inline code node
+                if insert_text_at.is_some() {
+                    let insert_at = insert_text_at.unwrap();
+
+                    // If there is still some collected text add it to he list of nodes to insert
+                    if !cur_text.is_empty() {
+                        nodes_to_add
+                            .insert(0, DomNode::new_text(cur_text.clone()));
+                    }
+
+                    // Insert the inline code node
+                    self.state.dom.insert(
+                        &insert_at,
+                        DomNode::new_formatting(
+                            InlineFormatType::InlineCode,
+                            nodes_to_add,
+                        ),
+                    );
+
+                    // Merge inline code nodes for clean up
+                    self.merge_formatting_node_with_siblings(&insert_at);
+                }
+            }
+            self.create_update_replace_all()
+        }
+    }
+
+    fn remove_and_clean_up_empty_nodes_until(
+        &mut self,
+        cur_handle: &DomHandle,
+        top_handle: &DomHandle,
+    ) {
+        self.state.dom.remove(cur_handle);
+        if cur_handle != top_handle
+            && self.state.dom.parent(cur_handle).children().is_empty()
+        {
+            self.remove_and_clean_up_empty_nodes_until(
+                &cur_handle.parent_handle(),
+                top_handle,
+            )
+        }
+    }
+
+    fn find_structure_ancestor(&self, handle: &DomHandle) -> Option<DomHandle> {
+        let parent = self.state.dom.parent(handle);
+        if parent.is_structure_node() {
+            Some(parent.handle().clone())
+        } else {
+            if parent.handle().has_parent() {
+                self.find_structure_ancestor(&parent.handle())
+            } else {
+                None
+            }
+        }
     }
 
     fn format_or_unformat(
```

Comment:
I'd prefer if utility functions like format_or_unformat were at the bottom, below your new tests.

Reply:
Sorry, I misread the diff! Ignore me.